pkg/minikube/config/translators.go

```diff
@@ -0,0 +1,172 @@
+package config
+
+import (
+	v162 "k8s.io/minikube/pkg/minikube/config/v162"
+)
+
+func tryTranslate(vcontran []VersionConfigTranslator) (interface{}, error) {
+
+	// Get the previous version translator
+	previousVersion := vcontran[0]
+	previousVersionConfig, err := previousVersion.TryLoadFromFile()
+	// if the translator couldn't load the config from the file, then it's probably even an older version, so let's recurse again deeper
+	if (err != nil || !previousVersion.IsValid(previousVersionConfig)) && len(vcontran) > 1 {
+		var err error
+		previousVersionConfig, err = tryTranslate(vcontran[1:])
+		if err != nil {
+			// Ah too bad, even the older versions couldn't translate it, this would bubble up to the end.
+			return nil, err
+		}
+	}
+	// Yes! The previous recurse iteration returned a successful and valid previousVersionConfig. Now let's translate it to the next version
+	translatedConfig, err := previousVersion.TranslateToNextVersion(previousVersionConfig)
+	return translatedConfig, err
+}
+
+var versionConfigTranslators = []VersionConfigTranslator{
+	{
+		TryLoadFromFile: func() (interface{}, error) {
+			return nil, nil
+		},
+		TranslateToNextVersion: func(config interface{}) (interface{}, error) {
+			return translateFrom163ToCurrent(config.(v162.MachineConfig))
+		},
+	},
+}
+
+type VersionConfigTranslator struct {
+	TryLoadFromFile        tryLoadFromFile
+	TranslateToNextVersion translateToNextVersion
+	IsValid                isValid
+}
+type translateToNextVersion func(interface{}) (interface{}, error)
+type tryLoadFromFile func() (interface{}, error)
+type isValid func(interface{}) bool
+
+func translateFrom163ToCurrent(oldConfig v162.MachineConfig) (*ClusterConfig, error) {
+	hypervUseExternalSwitch := (oldConfig.HypervVirtualSwitch != "")
+
+	return &ClusterConfig{
```

Comment:
Since the majority of the fields here are the same, instead of specifying each one individually you can:

1. Unmarshal the value of `oldConfig` to []byte
1. Marshal those contents into `newConfig`
1. Only manually set the fields that have changed in `newConfig` (like hypervUseExternalSwitch)

You could do that in any place where you're converting from an old config to a new config.